ff/src/biginteger/mod.rs

```diff
@@ -729,6 +729,38 @@ impl<B: Borrow<Self>, const N: usize> BitOr<B> for BigInt<N> {
     }
 }
 
+impl<const N: usize> Shr<u32> for BigInt<N> {
+    type Output = Self;
+
+    fn shr(self, mut rhs: u32) -> Self::Output {
```

Comment:
I thikn that it should be "(...) will NOT return an underflow error if the value being shifted is larger than N * 64", given that the limb size 64 for the case of BigInt.

I didn't document the trait definition because we are forcing the implementor to specify the `shr()` function. So, I didn't find an appropriate place to put this message. Let me know if it's fine like it is or if you have a proper place to put this message in the trait definition.